kit/Kit.cpp

```diff
@@ -2806,20 +2829,32 @@ int KitSocketPoll::kitPoll(int timeoutMicroS)
 
         // Flush at most maxEvents+1, or return when nothing left.
         _pollEnd = startTime + std::chrono::microseconds(timeoutMicroS);
+        bool isIDLE = false;
         do
         {
             int realTimeout = timeoutMicroS;
             if (_document && _document->needsQuickPoll())
                 realTimeout = 0;
 
-            if (poll(std::chrono::microseconds(realTimeout)) <= 0)
+            const int rc = poll(std::chrono::microseconds(realTimeout));
+            if(rc > 0)
+            {
+                // poll successful w/o timeout
+                const auto now = std::chrono::steady_clock::now();
+                timeoutMicroS =
+                    std::chrono::duration_cast<std::chrono::microseconds>(_pollEnd - now).count();
+            }
+            else if(rc == 0)
+            {
+                // poll timed out, i.e. IDLE client activity
+                isIDLE = true;
+                timeoutMicroS = 0;
+            }
+            else // poll error
                 break;
 
-            const auto now = std::chrono::steady_clock::now();
-            drainQueue();
+            drainQueue(isIDLE);
 
-            timeoutMicroS =
-                std::chrono::duration_cast<std::chrono::microseconds>(_pollEnd - now).count();
             ++eventsSignalled;
```

Comment:
This ++eventSignalled line wasn't incremented in the before-change case of a poll returning 0, but it is in the after-case (similarly timeoutMicroS is set to 0) and eventSignalled and timeoutMicroS are used after the loop, so that would be a change. Do we want that?

Would there be any real difference in intent, on, say, just passing false to the first drainQueue inside the loop and passing an isIDLE to the trailing drainQueue outside the loop, where isIDLE is set when poll is passed a non-zero timeout and returns 0 ?